Cannot build on Ubuntu 16.04

### Issue and Steps to Reproduce
I am trying to build lightning on a fresh installation of ubuntu 16.04 LTS
Build libwally and hsm failed with multiple linking errors aka "undefined reference to ..." 

gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609

### `getinfo` output

```
bip32.c:(.text+0xa23): undefined reference to `secp256k1_ec_pubkey_parse'
bip32.c:(.text+0xa64): undefined reference to `secp256k1_ec_pubkey_serialize'
external/libwallycore.a(libwallycore_la-bip32.o): In function `bip32_key_from_parent_path':
bip32.c:(.text+0x106a): undefined reference to `secp256k1_ec_pubkey_parse'
bip32.c:(.text+0x10af): undefined reference to `secp256k1_ec_pubkey_serialize'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_public_key_from_private_key':
sign.c:(.text+0x135): undefined reference to `secp256k1_ec_pubkey_serialize'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_public_key_decompress':
sign.c:(.text+0x21f): undefined reference to `secp256k1_ec_pubkey_parse'
sign.c:(.text+0x24d): undefined reference to `secp256k1_ec_pubkey_serialize'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_sig_normalize':
sign.c:(.text+0x345): undefined reference to `secp256k1_ecdsa_signature_parse_compact'
sign.c:(.text+0x357): undefined reference to `secp256k1_ecdsa_signature_normalize'
sign.c:(.text+0x365): undefined reference to `secp256k1_ecdsa_signature_serialize_compact'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_sig_to_der':
sign.c:(.text+0x43f): undefined reference to `secp256k1_ecdsa_signature_parse_compact'
sign.c:(.text+0x4a7): undefined reference to `secp256k1_ecdsa_signature_serialize_der'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_sig_from_der':
sign.c:(.text+0x583): undefined reference to `secp256k1_ecdsa_signature_parse_der'
sign.c:(.text+0x5aa): undefined reference to `secp256k1_ecdsa_signature_serialize_compact'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_sig_from_bytes':
sign.c:(.text+0x6fc): undefined reference to `secp256k1_ecdsa_signature_serialize_compact'
external/libwallycore.a(libwallycore_la-sign.o): In function `wally_ec_sig_verify':
sign.c:(.text+0x7ca): undefined reference to `secp256k1_ec_pubkey_parse'
sign.c:(.text+0x82e): undefined reference to `secp256k1_ecdsa_signature_parse_compact'
hsmd/hsm.o: In function `handle_ecdh':
/home/misha/lightning/hsmd/hsm.c:135: undefined reference to `secp256k1_ecdh'
hsmd/hsm.o: In function `bitcoin_pubkey':
/home/misha/lightning/hsmd/hsm.c:440: undefined reference to `secp256k1_ec_pubkey_parse'
hsmd/hsm.o: In function `sign_invoice':
/home/misha/lightning/hsmd/hsm.c:777: undefined reference to `secp256k1_ecdsa_sign_recoverable'
common/key_derive.o: In function `derive_revocation_key':
/home/misha/lightning/common/key_derive.c:162: undefined reference to `secp256k1_ec_pubkey_combine'
common/key_derive.o: In function `bip32_pubkey':
/home/misha/lightning/common/key_derive.c:259: undefined reference to `secp256k1_ec_pubkey_parse'
bitcoin/pubkey.o: In function `pubkey_from_der':
/home/misha/lightning/bitcoin/pubkey.c:15: undefined reference to `secp256k1_ec_pubkey_parse'
bitcoin/pubkey.o: In function `pubkey_to_der':
/home/misha/lightning/bitcoin/pubkey.c:25: undefined reference to `secp256k1_ec_pubkey_serialize'
bitcoin/pubkey.o: In function `secp256k1_pubkey_to_hexstr':
/home/misha/lightning/bitcoin/pubkey.c:69: undefined reference to `secp256k1_ec_pubkey_serialize'
bitcoin/signature.o: In function `signature_to_der':
/home/misha/lightning/bitcoin/signature.c:235: undefined reference to `secp256k1_ecdsa_signature_serialize_der'
bitcoin/signature.o: In function `signature_from_der':
/home/misha/lightning/bitcoin/signature.c:245: undefined reference to `secp256k1_ecdsa_signature_parse_der'
bitcoin/signature.o: In function `sig_valid':
/home/misha/lightning/bitcoin/signature.c:254: undefined reference to `secp256k1_ecdsa_signature_normalize'
wire/fromwire.o: In function `fromwire_secp256k1_ecdsa_signature':
/home/misha/lightning/wire/fromwire.c:125: undefined reference to `secp256k1_ecdsa_signature_parse_compact'
wire/fromwire.o: In function `fromwire_secp256k1_ecdsa_recoverable_signature':
/home/misha/lightning/wire/fromwire.c:140: undefined reference to `secp256k1_ecdsa_recoverable_signature_parse_compact'
wire/towire.o: In function `towire_pubkey':
/home/misha/lightning/wire/towire.c:53: undefined reference to `secp256k1_ec_pubkey_serialize'
wire/towire.o: In function `towire_secp256k1_ecdsa_signature':
/home/misha/lightning/wire/towire.c:75: undefined reference to `secp256k1_ecdsa_signature_serialize_compact'
wire/towire.o: In function `towire_secp256k1_ecdsa_recoverable_signature':
/home/misha/lightning/wire/towire.c:86: undefined reference to `secp256k1_ecdsa_recoverable_signature_serialize_compact'
collect2: error: ld returned 1 exit status
Makefile:280: recipe for target 'lightningd/lightning_hsmd' failed
make: *** [lightningd/lightning_hsmd] Error 1
rm external/libwally-core/src/secp256k1/libsecp256k1.la

```
`

--- a/Makefile
+++ b/Makefile
@@ -159,8 +159,8 @@ CWARNFLAGS := -Werror -Wall -Wundef -Wmissing-prototypes -Wmissing-declarations 
 CDEBUGFLAGS := -std=gnu11 -g -fstack-protector
 CFLAGS = $(CPPFLAGS) $(CWARNFLAGS) $(CDEBUGFLAGS) -I $(CCANDIR) $(EXTERNAL_INCLUDE_FLAGS) -I . -I/usr/local/include $(FEATURES) $(COVFLAGS) $(DEV_CFLAGS) -DSHACHAIN_BITS=48 -DJSMN_PARENT_LINKS $(PIE_CFLAGS)
 
-LDFLAGS = -L/usr/local/lib $(PIE_LDFLAGS)
-LDLIBS = -lm -lgmp -lsqlite3 $(COVFLAGS)
+LDFLAGS = $(PIE_LDFLAGS)
+LDLIBS = -L/usr/local/lib -lm -lgmp -lsqlite3 $(COVFLAGS)
 
 default: all-programs all-test-programs
 

